testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java

```diff
@@ -674,33 +692,244 @@ public void testOutputStats()
+                new OutputColumnMetadata("test_bigint", ImmutableSet.of(new ColumnDetail("tpch", "tiny", "orders", "totalprice"))));
+    }
+
+    @Test
+    public void testOutputColumnsWithAggregationAndHaving()
+            throws Exception
+    {
+        assertColumnLineage(
+                "SELECT min(orderstatus) as test_varchar, sum(totalprice) as test_bigint FROM orders GROUP BY custkey HAVING min(orderdate) > DATE '2000-01-01'",
+                new OutputColumnMetadata("test_varchar", ImmutableSet.of(new ColumnDetail("tpch", "tiny", "orders", "orderstatus"))),
+                new OutputColumnMetadata("test_bigint", ImmutableSet.of(new ColumnDetail("tpch", "tiny", "orders", "totalprice"))));
+    }
+
+    @Test
+    public void testOutputColumnsWithCountAll()
+            throws Exception
+    {
+        assertColumnLineage(
+                "SELECT orderstatus as test_varchar, count(*) as test_bigint FROM orders GROUP BY orderstatus",
+                new OutputColumnMetadata("test_varchar", ImmutableSet.of(new ColumnDetail("tpch", "tiny", "orders", "orderstatus"))),
+                new OutputColumnMetadata("test_bigint", ImmutableSet.of()));
+    }
+
+    @Test
+    public void testOutputColumnsWithWindowFunction()
+            throws Exception
+    {
+        assertColumnLineage(
+                "SELECT orderstatus as test_varchar, avg(totalprice) OVER (PARTITION BY custkey ORDER BY orderdate ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW) AS test_bigint FROM orders",
+                new OutputColumnMetadata("test_varchar", ImmutableSet.of(new ColumnDetail("tpch", "tiny", "orders", "orderstatus"))),
+                new OutputColumnMetadata(
+                        "test_bigint",
+                        ImmutableSet.of(
+                                new ColumnDetail("tpch", "tiny", "orders", "totalprice"),
+                                new ColumnDetail("tpch", "tiny", "orders", "custkey"),
+                                new ColumnDetail("tpch", "tiny", "orders", "orderdate"))));
+    }
+
+    @Test
+    public void testOutputColumnsWithPartialWindowClause()
+            throws Exception
+    {
+        assertColumnLineage(
+                "SELECT orderstatus as test_varchar, sum(totalprice) OVER (w ORDER BY custkey ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW) AS test_bigint FROM orders WINDOW w AS (PARTITION BY custkey)",
+                new OutputColumnMetadata("test_varchar", ImmutableSet.of(new ColumnDetail("tpch", "tiny", "orders", "orderstatus"))),
+                new OutputColumnMetadata(
+                        "test_bigint",
+                        ImmutableSet.of(
+                                new ColumnDetail("tpch", "tiny", "orders", "totalprice"),
+                                new ColumnDetail("tpch", "tiny", "orders", "custkey"))));
+    }
+
+    @Test
+    public void testOutputColumnsWithWindowClause()
```

Comment:
Please add a test where some output field does not depend on any source columns (e.g. a constant expression).

Reply:
Added a separate test for it - testOutputColumnsForSelectWithConstantExpression